packages/components/src/mobile/keyboard-aware-flat-list/use-text-input-offset.native.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import RCTAztecView from '@wordpress/react-native-aztec';
+import { useCallback } from '@wordpress/element';
+
+/** @typedef {import('@wordpress/element').RefObject} RefObject */
+/**
+ * Hook that calculates the currently focused TextInput's current
+ * caret Y coordinate position.
+ *
+ * @param {boolean}   scrollEnabled Whether the scroll is enabled or not.
+ * @param {RefObject} scrollViewRef ScrollView reference.
+ * @return {[Function]} Function to get the current TextInput's offset.
+ */
+export default function useTextInputOffset( scrollEnabled, scrollViewRef ) {
+	const getTextInputOffset = useCallback(
+		async ( caret ) => {
+			const { caretY = null } = caret ?? {};
+			const textInput =
+				RCTAztecView.InputState.getCurrentFocusedElement();
+
+			return new Promise( ( resolve ) => {
+				if (
+					scrollViewRef.current &&
+					textInput &&
+					scrollEnabled &&
+					caretY !== null
+				) {
+					textInput.measureLayout(
+						scrollViewRef.current,
+						( _x, y, _width, height ) => {
+							const caretYOffset =
+								// For cases where the caretY value is -1
+								// we use the y + height value, e.g the current
+								// character index is not valid or out of bounds
+								// see https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/TextKit/TextView.swift#L762
```

Comment:
It's likely best to use a [permalink](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet#linking-to-code), otherwise the referenced line of code may move or be removed.

Reply:
Thanks for the suggestion! Updated in https://github.com/WordPress/gutenberg/pull/48791/commits/a0806745f4b958e7e84802b4390570b7c2f472b8